AlphaWallet/Tokens/Coordinators/TokensCoordinator.swift

```diff
@@ -391,8 +391,12 @@ class TokensCoordinator: Coordinator {
 extension TokensCoordinator: TokensViewControllerDelegate {
     func didSelect(token: TokenObject, in viewController: UIViewController) {
         switch token.type {
-        case .ether:
-            delegate?.didPress(for: .send(type: .ether(config: session.config, destination: .none)), in: self)
+        case .ether, .xDai:
+            if token.name == "xDai" {
```

Comment:
Why not break up into 2 cases? :)

```
case .ether:
//...
case .xDai:
//...
```

Reply:
See the TODO, realm stores the users address in the database as an ethereum type, hence it never hits xDai here and I have to check it inside the same case.

Reply:
Hmm.. ok. A better way to do this if we don't do a migration: Push the logic into `TokenObject`, so `TokenObject.type` is always correct and we wouldn't forget to special case elsewhere.

```
    var type1: TokenType {
        get {
            if name == "xDai" {
                return .xDai
            } else {
                return TokenType(rawValue: rawType)!
            }
        }
        set {
            rawType = newValue.rawValue
        }
    }
```